File: src/Goods.Domain/CarCodes/CarCodes.cs
namespace Goods.Domain.Products;

public class CarCodes
{
    public Guid Id { get; set; }
    public String Code { get; set; }
    public Guid Regions { get; set; }

    public CarCodes
    (
        Guid id,
        String code,
        Guid regions
    )
    {
        Id = id;
        Code = code;
        Regions = regions;
    }
}

File: src/Goods.BackOffice/client-app/src/domain/carCodes/carCodes.ts
import { Page } from '../../tools/types/page';
import { Regions } from '../regions/regions';
import { mapToRegion } from '../regions/regions';
export class CarCodes {
	constructor(
		public readonly id: string,
		public readonly code: string,
		public readonly regions: Regions
	) { }
}

export function mapToCarCodesPage(data: any): Page<CarCodes> {
	return Page.convert(data, mapToCarCode);
}

export function mapToProducts(data: any[]): CarCodes[] {
	return data.map(mapToCarCode);
}

export function mapToCarCode(data: any): CarCodes {
	return new CarCodes(data.id, data.code, mapToRegion(data.regions));
}
